src/Lucene.Net/Search/Function/ReverseOrdFieldSource.cs

```diff
@@ -58,18 +58,18 @@ public class ReverseOrdFieldSource:ValueSource
 	{
 		private class AnonymousClassDocValues:DocValues
 		{
-			public AnonymousClassDocValues(int end, int[] arr, ReverseOrdFieldSource enclosingInstance)
+			public AnonymousClassDocValues(int end, IArray<int> arr, ReverseOrdFieldSource enclosingInstance)
```

Comment:
We could avoid boxing here by using `TArray where TArray : IArray<int>`

and during construction:
```csharp
			if (arr.GetType() == typeof(ManagedArray<int>))
				return new AnonymousClassDocValues<ManagedArray<int>>(end, (ManagedArray<int>)arr, this);
			
			return new AnonymousClassDocValues<UnmanagedArray<int>>(end, (UnmanagedArray<int>)arr, this);
```